### PR TITLE
added syntax support for begin isolation level

### PR DESCRIPTION
**Pull request by AmoebaProtozoa**
Signed-off-by: AmoebaProtozoa <davidyangad@gmail.com>

### What is changed and how it works?

Added parser support for `BEGIN ISOLATION LEVEL { SERIALIZABLE | REPEATABLE READ | READ COMMITTED | READ UNCOMMITTED }`

### Tests

`make test` passed

### CODE REVIEW

**Review by Orion7r — APPROVED** [review]
LGTM

File: parser.y [hunk 1]
```diff
@@ -2341,6 +2341,13 @@ BeginTransactionStmt:
 			Mode: ast.Optimistic,
 		}
 	}
+|	"BEGIN" "ISOLATION" "LEVEL" IsolationLevel
+	// supports for setting isolation level when starting transaction,
+	// currently it doesn't actually set anything
+	// TODO: Set isolation level before starting transaction
+	{
+		$$ = &ast.BeginStmt{}
+	}
```
Comment (pupillord) [inline parser.y:2350]: You should use a parameter in begin stmt to save isolation level.

**Review by pupillord — CHANGES_REQUESTED** [review]

File: parser.y (hunk 1 above)
Comment (AmoebaProtozoa) [reply parser.y]: It should now capture the correct isolation level

**Review by pupillord — APPROVED** [review]
LGTM

**Pull request merged**